[RFC][FSDP] Don't move ignored params / buffers to device

rohan-varma opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #108435
* __->__ #108033
* #108032

Since these are ignored by FSDP, don't move them.

Differential Revision: [D48727044](https://our.internmc.facebook.com/intern/diff/D48727044/)

rohan-varma committed:
[RFC][FSDP] Don't move ignored params / buffers to device

Since these are ignored by FSDP, don't move them.

Differential Revision: [D48727044](https://our.internmc.facebook.com/intern/diff/D48727044/)

[ghstack-poisoned]

awgu commented on torch/distributed/fsdp/_init_utils.py:
```diff
@@ -24,6 +24,7 @@
 import torch.distributed.fsdp._traversal_utils as traversal_utils
 import torch.distributed.fsdp.fully_sharded_data_parallel as fsdp_file
 import torch.nn as nn
+from torch import Tensor
```
nit: If we do this, I feel like we should replace all `torch.Tensor` in the file with `Tensor`, or else it gets fragmented.

Maybe just use `torch.Tensor` on L857?

awgu commented on torch/distributed/fsdp/_init_utils.py:
```diff
@@ -884,7 +898,9 @@ def _move_module_to_device(
         # NOTE: This includes moving ignored modules' parameters. If we
```
We should remove this note.

awgu commented:
I am on board with the change.

rohan-varma committed:
Update on "[RFC][FSDP] Don't move ignored params / buffers to device"


Since these are ignored by FSDP, don't move them.

Differential Revision: [D48727044](https://our.internmc.facebook.com/intern/diff/D48727044/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[RFC][FSDP] Don't move ignored params / buffers to device"


Since these are ignored by FSDP, don't move them.

Differential Revision: [D48727044](https://our.internmc.facebook.com/intern/diff/D48727044/)

[ghstack-poisoned]

awgu commented:
Please check lint before landing. Thanks!

rohan-varma committed:
Update on "[RFC][FSDP] Don't move ignored params / buffers to device"


Since these are ignored by FSDP, don't move them.

Differential Revision: [D48727044](https://our.internmc.facebook.com/intern/diff/D48727044/)

[ghstack-poisoned]

rohan-varma committed:
Update on "[RFC][FSDP] Don't move ignored params / buffers to device"


Since these are ignored by FSDP, don't move them.

Differential Revision: [D48727044](https://our.internmc.facebook.com/intern/diff/D48727044/)

[ghstack-poisoned]

ppwwyyxx commented on torch/distributed/fsdp/_init_utils.py:
```diff
@@ -841,6 +851,7 @@ def _get_modules_to_materialize(
+    ignored_buffers: Set[torch.Tensor],
     device_from_device_id: Optional[torch.device],
 ) -> None:
     """
```
the docstring below should be updated
